spec/mailers/user_mailer_spec.rb

```diff
@@ -0,0 +1,47 @@
+#
+# frozen_string_literal: true
+
+# spec/mailers/user_mailer_spec.rb
+require 'rails_helper'
+
+RSpec.describe UserMailer, type: :mailer do
+  describe 'my_email' do
+    let(:user) { User.create(email: Faker::Internet.email) }
+
+    context 'when the user language is set to English' do
+      before do
+        @original_locale = I18n.locale
+        I18n.locale = :'en-CA'
+      end
+
+      after do
+        I18n.locale = @original_locale
+      end
+
+      let(:mail) { UserMailer.welcome_notification(user) }
+
+      it 'renders the email subject in English' do
+        expect(mail.subject).to include('Welcome to')
```

Comment:
@200455939-yashu If we know the full string of the subject, we could check for equality. 

That is 

```
expect(mail.subject).to eq('Welcome to REST OF WELCOME SUBJECT')
```

We can do the same with the French version. And we don't need to check for the ```not_to``` version.

Reply:
On this, some of the tests were breaking the PR request. There are some places it was showing ("Welcome to Foo") and some instances ("Bienvenue sur DMP") instead of ("Bienvenue sur PGD").

Reply:
```Welcome to Foo``` is probably due to the configuration of ```Foo``` as the default organization name.

As for the rest, I believe it should be ```Bienvenue sur PGD```. ```Bienvenue sur DMP``` is mixing English and French. Are these texts from subjects appearing randomly?

Reply:
Yes. Texts are very Random.

Reply:
Updated